harness/tests/experiment/pytorch/test_deepspeed_autotuning.py

```diff
@@ -0,0 +1,206 @@
+# type: ignore
+import copy
+import json
+import os
+import pathlib
+import tempfile
+from argparse import Namespace
+from typing import Any, Dict, Iterator, List, Optional, Sequence, Union
+from unittest import mock
+
+from determined import searcher
+import determined.pytorch.deepspeed as det_deepspeed
+from deepspeed.runtime import config_utils
+from determined.common.api import bindings
+
+from determined import workload
+from tests.experiment import utils  # noqa: I100
+from tests.experiment.fixtures import deepspeed_linear_model
+from tests.custom_search_mocks import SimulateMaster, MockMasterSearchRunner
+
+
+######
+
+from determined.pytorch.deepspeed.dsat import autotune, _utils, _defaults
+from determined.pytorch.deepspeed.dsat import DSATTrialTracker, DSATTrial
+from determined.pytorch.deepspeed.dsat._dsat_search_method import (
+    SimpleBatchSearchMethod,
+    DSATRandomSearchMethod,
+)
+
+
+BASE_EXPERIMENT_FIXTURE_PATH = (
+    pathlib.Path(__file__).resolve().parent.parent.joinpath("fixtures/deepspeed_autotune")
+)
+
+MODEL_INFO_PROFILE_METRIC_FIXTURE = {
+    "num_params": 60192808,
+    "trainable_num_params": 60192808,
+    "activation_mem_per_gpu": 89828352,
+    "rank": 0,
+    "gpu_mem": 15843721216,
+}
+
+
+@mock.patch("determined.experimental.client.create_experiment")
+def test_autotuning_module_transforms(
```

Comment:
A little confused about what this is testing. Is it _only_ testing that the searcher name is `single`? Thrown off by the `module_transforms` in the test name.

Reply:
Namely, our `autotune` python module, whose sole job is to take the user config, transform it into the "single" custom searcher we run to orchestrate the rest of the runs. It is also implicitly testing that these dictionaries merge and that we call a remote searcher for "create_experiment" in the module. I don't know how much work we actually want this module to do in the long term, so I kept it to a minimum for now